os/drivers/audio/syu645b.c

```diff
@@ -1031,19 +1040,201 @@ static void syu645b_hw_reset_config(FAR struct syu645b_dev_s *priv)
+		pos--;
+
+		if (decimal == 0) {
+			if (pos > 1) {
+				for(int i=2; i<=pos; i++){
+					hexadecimal[pos] = '0';
+				}
+			}
+			break;
+		}
+	}
+
+	return 0;
+}
+
+static int string_to_script(char *string, t_codec_init_script_entry *script){
+	int cnt = 0;
+
+	const char *ptr = string;
+	while (*ptr && cnt < SYU645B_REG_DATA_TYPE_MAX) {
+		if (*ptr == '0' && (*(ptr + 1) == 'x' || *(ptr + 1) == 'X')) {
+			char *end_ptr;
+			uint8_t val = strtol(ptr, &end_ptr, 16);
+
+			if (ptr != end_ptr) {
+				if (cnt == 0) {
+					script->addr = val;
+				}
+				else {
+					script->val[cnt-1] = val;	
+				}
+				ptr = end_ptr;
+				cnt++;
+				continue;
+			}
+		}
+		ptr++;
+	}
+
+	script->delay = 0;
+	script->type = cnt;
+
+	return 0;
+}
+
 static void syu645b_set_equalizer(FAR struct syu645b_dev_s *priv, uint32_t preset)
 {
 	/* Reset First */
 	syu645b_reset_config(priv);
 
-	/* And then Set default EQ Set */
+	/* Initialize for i2c write */
+	uint8_t reg[SYU645B_REG_DATA_TYPE_MAX];
+	FAR struct i2c_dev_s *dev = priv->i2c;
+	FAR struct i2c_config_s *syu645b_i2c_config = &(priv->lower->i2c_config);
+	
+	int ret = 0;
+
+	/* Parse equlizer json to gJSON */
+	ret = parse_equalizer_metadata_json();
+	if (ret) {
+		printf("[TEST] goto default_script parse_equalizer_metadata_json\n");
+		goto default_script;
+	}
+
+	/* Convert preset hex to string for key */
+	char c_preset[11];
+	ret = dec_to_hex_string(preset, c_preset);
+	if (ret) {
+		goto default_script;
+	}
+
+	/* Executes given script through i2c to configure SYU645B device. */
+	cJSON *search = cJSON_GetObjectItemCaseSensitive(gJSON, c_preset);
+
+	/* Find proper key */
+	if (search){
+		cJSON *key;
+		t_codec_init_script_entry script;
+		
+		cJSON_ArrayForEach(key, search){
+			if (strcmp(key->string, "_comment") == 0) {
+				printf("[TEST] find _comment\n");
+				continue;
+			}
+			printf("[TEST] script | %s : %s\n", key->string, key->valuestring);
+
+			ret = string_to_script(key->valuestring, &script);
+			if (ret) {
+				goto default_script;
+			}
+
+			if (script.addr == 0x08 && script.type == SYU645B_REG_D_2BYTE) {
+				priv->max_volume = script.val[0];
+				printf("[TEST] priv->max_volume update %d \n", priv->max_volume);
+			}
+
+			reg[0] = script.addr;
+			for (int i = 0; i < script.type-1; i++) {
+				reg[i+1] = script.val[i];
+			}
+			printf("[TEST] addr : %x / val : ", script.addr);
+			for(int i=0; i<script.type-1; i++){
+				printf("%x ", script.val[i]);
+			}
+			printf("\n");
+
+			ret = i2c_write(dev, syu645b_i2c_config, (uint8_t *)reg, script.type);
+			if (ret < script.type) {
+				auddbg("Error, cannot write to reg addr 0x%x, ret = %d\n", script.addr, ret);
+				break;
+			}
+			if (script.delay > 0) {
+				up_mdelay(script.delay);
+			}
+		}
+		
+		/* And then setVolume Again */
+		if (!priv->mute) {
+			syu645b_setvolume(priv);
+			return 0;
+		}
+	}
+	/* Not find proper key, execute default script*/
+	else {
+		goto default_script;
+	}
+
+default_script:
```

Comment:
During init, we call set_equalizer(0);
I think if we provide function like a set_basic_equalizer, then set_equalizer(0) will not execute above logic.
Also we can remove 'goto default_script'
goto is very useful actually but to prevent human error, we should use very rarely.